Extensions/NotificationService/NotificationService.swift

```diff
@@ -2,34 +2,102 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+import Shared
+import Storage
+import Sync
 import UserNotifications
 
 class NotificationService: UNNotificationServiceExtension {
-
-    var contentHandler: ((UNNotificationContent) -> Void)?
-    var bestAttemptContent: UNMutableNotificationContent?
+    var display: SyncDataDisplay!
+    var profile: Profile!
 
     override func didReceive(_ request: UNNotificationRequest, withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
-        self.contentHandler = contentHandler
-        bestAttemptContent = (request.content.mutableCopy() as? UNMutableNotificationContent)
 
-        var userInfo = request.content.userInfo
-        userInfo["url"] = "https://mozilla.org"
+        guard let content = (request.content.mutableCopy() as? UNMutableNotificationContent) else {
+            return
+        }
+
+        let userInfo = request.content.userInfo
         NSLog("NotificationService APNS NOTIFICATION \(userInfo)")
-        if let bestAttemptContent = bestAttemptContent {
-            // Modify the notification content here...
-            bestAttemptContent.title = "\(bestAttemptContent.title) [modified]"
-            bestAttemptContent.userInfo = userInfo
-            contentHandler(bestAttemptContent)
+
+        self.display = SyncDataDisplay(content: content, contentHandler: contentHandler)
+        self.profile = ExtensionProfile(localName: "profile", delegate: display)
+
+        let handler = FxAPushMessageHandler(with: profile)
+
+        handler.handle(userInfo: userInfo).upon {_ in
+            self.display.displayNotification(true)
```

Comment:
I think the notification was just blank in our testing. How can we say that a tab arrived or something like that? (not sure if we have access to the info about which tab really arrived)